### PR TITLE
Chore/nft list update UI

### PR DESCRIPTION
**Pull request by haongo138**
**What does this PR do?**

-   Resolve following UI issues 
image

- UI previews  
image
image

### CODE REVIEW

File: src/commands/community/nft/list.ts [hunk 1]
```diff
@@ -97,29 +62,39 @@ async function renderSupportedNFTList(collectionList: NFTCollection[]) {
...
+    } else {
+      collectionName =
+        handleTextOverflow(ctx, cName, 80) +
+        ` (${handleTextOverflow(ctx, item.symbol?.toUpperCase(), 200)})`
+    }
+
     const chainName = item?.chain?.name
       ? handleTextOverflow(ctx, item?.chain?.name, 320)
-      : "No data"
+      : "TBD  "
```
Comment (anhnh12) [inline src/commands/community/nft/list.ts:97]: trim space

File: src/commands/community/nft/list.ts [hunk 2]
```diff
@@ -139,18 +114,32 @@ async function renderSupportedNFTList(collectionList: NFTCollection[]) {
...
     // chain name
+    const rectHeight = 40
+    const rectWidth = widthOf(ctx, chainName)
     const chainNameConf = {
       x: rowIn1stCol.x,
       y: clt1stColConf.y + clt1stColConf.mb,
       mb: 20,
     }
 
+    const col1stRectStats: RectangleStats = {
```
Comment (anhnh12) [inline src/commands/community/nft/list.ts:125]: 2 same code blocks -> please generalize this
the only different thing between these 2 loops is `x`, please use loop index to adjust correct x
e.g. 
```
i % 2 === 0 -> x = 0
i % 2 === 1 -> x = 400
```

File: src/commands/community/nft/list.ts [hunk 3]
```diff
@@ -139,18 +114,32 @@ async function renderSupportedNFTList(collectionList: NFTCollection[]) {
...
+      x: {
+        from: chainNameConf.x,
+        to: chainNameConf.x + rectWidth - 6,
+      },
+      y: {
+        from: clt1stColConf.y + 20,
```
Comment (anhnh12) [inline src/commands/community/nft/list.ts:131]: `clt1stColConf` -> vars name should be readable

File: src/commands/community/nft/list.ts [hunk 4]
```diff
@@ -240,7 +241,8 @@ async function composeNFTListEmbed(msg: Message, pageIdx: number) {
 
   const totalPage = Math.ceil(total / size)
   const embed = composeEmbedMessage(msg, {
-    image: "attachment://nftlist.png",
+    title: `${getEmoji("heart")} Supported NFT Collections`,
```
Comment (anhnh12) [inline src/commands/community/nft/list.ts:244]: try using `author` instead of title and observe

File: src/utils/canvas.ts [hunk 5]
```diff
@@ -336,3 +336,23 @@ export async function renderChartImage({
...
+  c: CanvasRenderingContext2D,
+  text: string,
+  maxWidth: number
+): string {
+  let width = c.measureText(text).width
+  const ellipsis = "…"
+  const ellipsisWidth = c.measureText(ellipsis).width
+  if (width <= maxWidth || width <= ellipsisWidth) {
+    return text
+  } else {
```
Comment (anhnh12) [inline src/utils/canvas.ts:350]: redundant else

File: src/utils/discordEmbed.ts [hunk 6]
```diff
@@ -313,12 +314,22 @@ export function listenForPaginateAction(
...
-        components: getPaginationRow(page, +totalPage),
-      })
+
+      if (!withAttachmentUpdate) {
+        await replyMsg.edit({
+          embeds,
+          components: getPaginationRow(page, +totalPage),
+        })
+      } else {
+        await replyMsg.removeAttachments()
```
Comment (anhnh12) [inline src/utils/discordEmbed.ts:326]: check if files are not undefined/null/empty?

**Comment by anhnh12** [comment]
have you filtered out collections with empty name?

**Comment by anhnh12** [comment]
reduce image height

File: src/utils/canvas.ts (hunk 5 above)
Comment (haongo138) [reply src/utils/canvas.ts:350]: i dont really understand this comment, can u recommend me a way to fix this

**Comment by haongo138** [comment]
>

**Pull request merged**